### PR TITLE
Issue #1107 - add support for Identifier pattern value constraints

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java [hunk 1]
```diff
@@ -150,7 +148,7 @@ public int getMaxArity() {
 
     private boolean contains(Map<String, Set<String>> codeSetMap, Coding coding) {
         String system = (coding.getSystem() != null) ? coding.getSystem().getValue() : null;
-        String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : FHIRRegistry.getInstance().getLatestVersion(system, CodeSystem.class);
+        String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : null;
```
Comment (lmsurpre) [inline fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java:151]: is this change related?

File: fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java [hunk 2]
```diff
@@ -304,13 +313,38 @@ private String generatePatternValueConstraint(ElementDefinition elementDefinitio
...
+                    sb.append(" <= 1");
+                } else {
+                    sb.append(" = 1");
+                }
+            } else {
+                sb.append(" >= ").append(min).append(" and ").append(identifier).append(".where(system = '").append(system).append("').count() <= ").append(max);
+            }
+
+            if (!node.children.isEmpty()) {
+                sb.append(" and (").append(identifier).append(".where(system = '").append(system).append("').exists()").append(" implies (").append(identifier).append(".where(system = '").append(system).append("' and ").append(generate(node.children)).append(")))");
```
Comment (lmsurpre) [inline fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java:346]: any way to format this to make it a little simpler to see whats going on?

Comment (lmsurpre) [reply fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java:346]: or maybe just a comment to indicate what this and clause is doing (at a high level)?

Comment (JohnTimm) [reply fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java:346]: The way the constraint is generated, there isn't much in the way of indentation that will do it justice. Looking at the actual generated constraint is more helpful.

File: fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java (hunk 1 above)
Comment (JohnTimm) [reply fhir-path/src/main/java/com/ibm/fhir/path/function/MemberOfFunction.java:151]: Only tangentially related. There were warnings associated with a bug in the CARIN BB IG because we were defaulting an unknown version to Latest Version was causing superfluous warnings to be generated. This change makes the code a bit more tolerant.
